public/docs/_examples/cb-validation/ts/app/hero-form-model-binding.component.html

```diff
@@ -0,0 +1,75 @@
+ <!-- #docplaster -->
+ <!-- #docregion -->
+<div class="container">
+  <div [hidden]="submitted">
+    <h1>Hero Form (Model-Driven with Data Binding)</h1>
+    <form [ngFormModel]="heroForm" *ngIf="active" 
+          (ngSubmit)="onSubmit()">
+      <div class="form-group">
+        <!-- #docregion name-with-error-msg -->
+        <label for="name">Name</label>
+        <input type="text" class="form-control" 
+               [(ngModel)]="model.name"
+               ngControl="name"
+               [ngClass]="{'required': isRequired('name')}">
```

Comment:
If possible, I'd be inclined to include the `minlength` and `maxlength` attrs here too for the built-in browser behavior. Otherwise this technique isn't actually equivalent to the template-driven approach.